mantidproject/mantid
# Matplotlib 3.8 update

### Description of work

#### Summary of work
 Update Matplotlib to 3.8.4 from 3.7

 
This pr updates the codebase of Mantid to use a newer version of Matplotlib to keep up to date with new features.

### To test:
1. Run the following tests:
   `FindPeakAutomaticTest`, `test_figureIntegration`, `test_toolbar`, `test_figuretype`, `mantidaxesTest`
2. Try to recreate the issues to see if they occur. E.g The issue with the mantidaxes was caused when a plot was plotted through the script editor, and another plot created from data was dragged onto that plot.
3.  Read my code, check for any cases I might have not thought of.

Note: getting the current colour for the `prop_cycler` has been completely removed and therefore that might be the only way of doing it even though it is not foolproof.

 

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
